rmgpy/solver/base.pyx

```diff
@@ -1229,6 +1229,12 @@ cdef class ReactionSystem(DASx):
                         logging.info('At time {0:10.4e} s, reached target termination RateRatio: {1}'.format(self.t,charRate/maxCharRate))
                         self.logConversions(speciesIndex, y0)
                         
+                elif isinstance(term, TerminationAvgMW):
+                    avgMW = numpy.dot(self.y[:numCoreSpecies],self.mw[:numCoreSpecies])*constants.Na*1000.0/self.y[:numCoreSpecies].sum()
```

Comment:
What I mean is here, you are just using number average termination criteria. Do you think about giving users options to choose from number average molecular weight sum(y*M)/sum(y) and weight average molecular weight sum(y*M^2)/sum(y*M). Or even add a new termination criterion like the polydispersity index. I've heard they are important parameters but not quite clear about how people use them. I am totally fine if you don't want to add those for now, but I am happy to hear what do you think?

Reply:
So in general there are two major considerations for termination criteria:  
1) It needs to trigger at useful times
2) A user needs to have some idea what to set it to, either it needs to have a physical meaning to them or they need to be able to use a default setting

In this case I chose number averaged molecular weight because it is the simplest to calculate, of these.  There are definitely use cases in polymers for Weight averaged molecular weight based termination, so I can add that, although it's bit more complicated for a user to set.  

Polydispersity index however, likely would not make a good termination criterion because it wouldn't trigger at physically meaningful times unless you could guess what the limiting value and general trend were for the polymer before running RMG, if you guess slightly wrong your run could terminate way too early or way too late.

Reply:
Thanks for your great explanation. I am okay with the current PR. For weight average molecular weight, it may need more specific cases to decide: (1) whether useful, (2) whether used simultaneously with number average molar weight, etc. Anyway, thanks for the updates and efforts.